src/apps/filters/h3.c

```diff
@@ -727,6 +727,557 @@ SUBCOMMAND(localIjToCell,
     return E_SUCCESS;
 }
 
+/// Hierarchical subcommands
+
+SUBCOMMAND(cellToParent,
+           "Returns the H3 index that is the parent (or higher) of the "
+           "provided cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 0-14 inclusive, excluding 15 as it can "
+                      "never be a parent"};
+    Arg *args[] = {&cellToParentArg, &helpArg, &cellArg, &resArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index parent;
+    int valid = H3_EXPORT(isValidCell)(cell);
+    if (valid == 0) {
+        return E_CELL_INVALID;
+    }
+    H3Error err = H3_EXPORT(cellToParent)(cell, res, &parent);
+    if (err) {
+        return err;
+    }
+    h3Println(parent);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(cellToChildren,
+           "Returns a JSON array of H3 indexes that are the children (or "
+           "lower) of the provided cell") {
+    // TODO: This function contains a lot of code that is very similar to
+    // `gridDisk`. If this happens again we should DRY them.
+    DEFINE_CELL_ARG(cell, cellArg);
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 1-15 inclusive, excluding 0 as it can "
+                      "never be a child"};
```

Comment:
could cellToChildren be called with res 0 if the input cell is res 0? It would only return the input cell.